### DISCUSSION TITLE
Cannot connect you node.js demo app

### DISCUSSION THREAD

**Discussion by ramiel** [planetscale/discussion#127] [Bug Reports] [answered]
I'm having issue connecting to db in my node.js app. So I decided to run you example project (https://github.com/planetscale/express-example). I run the project like this

```
DATABASE_URL='mysql://<USER>:<PWD>@<HOST>/<DBNAME>?sslmode=require' node app.js 
```

with the values given by the platform of course. I get this error

```
/projects/express-example/node_modules/mysql/lib/protocol/Parser.js:437
      throw err; // Rethrow non-MySQL errors
      ^

Error: ER_UNKNOWN_ERROR: unknown error: Code: UNAVAILABLE
server does not allow insecure connections, client must use SSL/TLS

    at Handshake.Sequence._packetToError (/home/fabrizio/Modelli/Projects/experiments/express-  
   [... other stack trace....]
 {
  code: 'ER_UNKNOWN_ERROR',
  errno: 1105,
  sqlMessage: 'unknown error: Code: UNAVAILABLE\n' +
    'server does not allow insecure connections, client must use SSL/TLS\n',
  sqlState: 'HY000',
  fatal: true
}
```

**Comment by johnyerhot** [comment, marked as answer]
👋  @ramiel  - try flipping `sslmode=require` to `ssl=true`. So something like this for the connection string:

`DATABASE_URL='mysql://<USER>:<PWD>@<HOST>/<DBNAME>?ssl=true`

The connection string you mentioned is meant to be running through the proxy set up by our CLI (and also generated by the CLI) and it looks like you're attempting to connect directly. Node should pick up your system's certs and whatnot and create a TLS connection with `ssl=true` that way.

Let me know if you hit any snags there!

EDIT: @DanielOberlechner wrote about his experience on MacOS Ventura + Prisma [below](https://github.com/planetscale/discussion/discussions/127#discussioncomment-3070954). It's been helpful for other Mac users.

**Comment by ramiel** [comment]
Thanks @johnyerhot this is working with driver `mysql`, but it's not working with `mysql2`. I'll find the way

**Reply by jvelo** [reply]
@ramiel found a little trick for `mysql2` : you can pass `?ssl={}` instead

**Reply by jgreet** [reply]
Right, it looks like passing any object for `ssl` will enable SSL in `mysql2`, and once it's enabled the defaults should work.

So for an env variable you could use something like:
```
DATABASE_URL=mysql://<USER>:<PASSWORD>@<HOST>/<DB-NAME>?ssl={} 
```

Or to configure it in code:

```js
const mysql = require('mysql2');

const connection = mysql.createConnection({
  user: '********',
  password: 'pscale_pw_********',
  database: 'your-db-name',
  host: '*******.your-region.psdb.cloud',
  ssl: {}
});
```

**Reply by sean-tmg** [reply]
Please can we get this information on the website. Was a real head cracker for me for a while.

**Reply by dgurns** [reply]
Yep this worked for me. Super obscure - thanks for the help.

**Reply by blobinabottle** [reply]
Go this today, still very hard to find that valuable information. Copy/pasting using the connect with prisma URL from the pscale dashboard was just not working...

**Comment by elliotjreed** [comment]
If anyone is using Sequalize, I was able to connect using:

```javascript
const connection = new Sequelize("mysql://<USER>:<PASSWORD>@<HOST>/<DB-NAME>", { dialect: "mysql", dialectOptions: { ssl: {} } });
```

**Reply by HazeerZacky** [reply]
WORKS!!!

**Comment by DanielOberlechner** [comment]
Hey guys,
I've got this connection string using the scaleplanet website:
DATABASE_URL='mysql://USER:pscale_pw_PASSWORD@SERVER/DATABASE?ssl={"rejectUnauthorized":true}'
but I still get the following error:
```
Error: Migration engine error:
unknown error: Code: UNAVAILABLE
server does not allow insecure connections, client must use SSL/TLS
```

I tried all the suggestions here from this issue but nothing seemed to be working for me.

Has anyone a hint what I could do or what is wrong? I'm using mysql2 by the way.

Kind regards,
Daniel Oberlechner

Now I found this on google:
https://docs.planetscale.com/concepts/secure-connections

When I try to do this on macOS Ventura installed mysql via brew everything went smoothly but then If I try to enter this command:
mysql --ssl-mode=VERIFY_IDENTITY --ssl-ca=/etc/ssl/cert.pem

I get the following error:
ERROR 2026 (HY000): SSL connection error: error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed

What can I do now?!

After a while I finally found the solution:
Generated by the website from Prisma WRONG:
DATABASE_URL='mysql://USER:pscale_pw_PASSWORD@SERVER/DATABASE?ssl={"rejectUnauthorized":true}'

Generated by me after some googling which works on macOS Ventura WORKS:
DATABASE_URL='mysql://USER:pscale_pw_PASSWORD@SERVER/DATABASE?sslcert=/etc/ssl/cert.pem'

Leave a like if I could help someone out there!
Daniel Oberlechner

**Reply by CaptainYarb** [reply]
Can we please get an update to the set answer for Mac users @johnyerhot ? This was a blocker for me as well.

**Reply by johnyerhot** [reply]
Can do!

**Reply by dcsan** [reply]
(two months later)
just going thru setup now, and it's bizarre the tutorial requires this much obscure googling and comments from frustrated users

**Reply by DanielOberlechner** [reply]
@dcsan looks like Prisma is still quite young software and therefore some googling is required ... :)

**Reply by lamalamaMark** [reply]
This fixed my local issue on Mac Ventura.

**Reply by johnblythe** [reply]
incredibly helpful, ty!

**Comment by Gilpop8663** [comment]
## Error Code

```
PrismaClientInitializationError: 
Invalid `prisma.idea_user.create()` invocation:
Error querying the database: Server error: `ERROR HY000 (1105): unknown error: Code: UNAVAILABLE
server does not allow insecure connections, client must use SSL/TLS
```

In my case, the model's variable name was the problem

I am currently using Prisma, mysql, PlanetScale, and NextJs and deploy through vercel

and my DATABASE_URL = mysql://USER:PASSWORD@LOATION.connect.psdb.cloud/APPNAME?sslaccept=strict

I thought the database was wrong, so I did it for two days, but I found out the model name was wrong


BEFORE

![image](https://user-images.githubusercontent.com/80146176/191468600-4cd4138b-7327-4076-bf28-ecb0405f2a28.png)


![image](https://user-images.githubusercontent.com/80146176/191470058-9a551446-50e5-4251-aaa7-27609e45086d.png)


![image](https://user-images.githubusercontent.com/80146176/191469490-3296ecc0-0bcf-4554-b9b0-bcb25ede3b32.png)


-----

AFTER

![image](https://user-images.githubusercontent.com/80146176/191468698-44333ea2-e4a2-4ea3-a8e4-d0a6b453d8c3.png)


![image](https://user-images.githubusercontent.com/80146176/191469221-5ea050f1-ed4a-4039-938b-5f9ef0d3840b.png)


# ~~Prisma model name should only use lowercase letters!!~~ 
# SnakeCase is also possible && Underscore is impossible !


![image](https://user-images.githubusercontent.com/80146176/191473601-ada6303f-1c5e-447e-a42d-093fe14965c3.png)

**Reply by dcsan** [reply]
in prisma you can also do this, haven't tested if it helps...

`@@map("table_name")`
